google/resource_container_cluster.go

```diff
@@ -504,13 +514,15 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 		locationsList := v.(*schema.Set).List()
 		locations := []string{}
 		for _, v := range locationsList {
-			location := v.(string)
-			locations = append(locations, location)
-			if location == zoneName {
-				return fmt.Errorf("additional_zones should not contain the original 'zone'.")
+			loc := v.(string)
```

Comment:
Consider simplifying this by using convertStringSet and schema.Set#Contains.

Also I am not sure why we have this block below:

```go
if isZone(location) {
   locations = append(locations, location)
}
```
Why are we adding the "main" zone to the list of additional zones... Consider adding a comment to clarify this.